### PR TITLE
Test on  Ruby Versions  3.1 /3.0 

### PR DESCRIPTION
**Pull request by poorndm**
IPACK-67 chef/mixlib-versioning -  General purpose Ruby library that allows you to parse, compare, and manipulate version strings in multiple formats.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by poorndm** [comment]
@tas50 
In [.expeditor/verify.pipeline.yml]
> @@ -44,7 +44,27 @@ steps:
   expeditor:
     executor:
       docker:
-        image: ruby:2.7-buster
+        image: ruby:2.7
+
+- label: run-lint-and-specs-ruby-3.0
+  command:
+    - apt-get update
+    - apt-get install -y libarchive-dev
        
 I have copied the same for 3.0/ 3.1 versions  as what you did for the older ruby versions . 
Please do let me know If I should remove it for all versions .

File: .expeditor/verify.pipeline.yml [hunk 1]
```diff
@@ -6,45 +6,73 @@ expeditor:
 
 steps:
 
-- label: run-lint-and-specs-ruby-2.4
+- label: run-lint-ruby-2.4
+  command:
+    - apt-get update
```
Comment (jeremiahsnapp) [inline .expeditor/verify.pipeline.yml:11]: You can remove the `apt-get update` and `apt-get install` commands for the lint test.

File: .expeditor/verify.pipeline.yml [hunk 2]
```diff
@@ -6,45 +6,73 @@ expeditor:
...
     - .expeditor/run_linux_tests.sh rake
   expeditor:
     executor:
       docker:
-        image: ruby:2.7-buster
+        image: ruby:2.7
+
+- label: run-specs-ruby-3.0
+  command:
+    - apt-get update
```
Comment (jeremiahsnapp) [inline .expeditor/verify.pipeline.yml:61]: Please remove `apt-get update` for ruby 3.0 and 3.1 tests since you aren't running `apt-get install`.

**Review by tas50 — APPROVED** [review]

File: Rakefile [hunk 3]
```diff
@@ -40,4 +40,4 @@ task :console do
   IRB.start
 end
 
-task default: %i{spec style}
+task default: %i{spec}
```
Comment (tas50) [inline Rakefile:43]: Instead of modifying the default task here it would be better to just change the expeditor config to use `rake spec`.

**Review by tas50 — APPROVED** [review]

**Pull request merged**